design/service-binding.md

```diff
@@ -0,0 +1,851 @@
+    owned:
+    - description: Represents a Kafka cluster
+      displayName: Kafka
+      kind: Kafka
+      name: kafkas.kafka.strimzi.io
+      version: v1beta1
+      statusDescriptors:
+      - description: The binding information for Kafka clients
+        displayName: ConfigMap containing binding information
+        path: status.bindingConfigMap
+        x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap
+```
+
+#### Consuming client's ServiceBindingRequest
+
+All of the required annotations are applied to the `Kafka` CSV, so the binding should only need to refer to the `Kafka` CR.
+
+``` yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: my-binding
+spec:
+  services:
+  - group: kafka.strimzi.io
+    kind: Kafka
+    version: v1beta1
+    resourceRef: my-cluster
+```
+
+The consuming client needs to know the listener name.
+
+The Service Binding Operator creates a `ConfigMap` with the same content as the Strimzi binding ConfigMap.
+
+## Binding to a Kafka cluster with TLS but no authentication
+
+The addition with this scenario is that Kafka clients need access to the CA certificate that signed the broker's server certificate. The Service Binding Specification does not currently have support for a separate CA certificate, which seems like a simple enhancement, which it indicated using `caSecret` in the example below.
+
+The CA certificate is most easily obtained from the `<cluster>-cluster-ca-cert` secret using the `ca.p12` and `ca.password` fields in the secret. To enable the Service Binding Operator to obtain this information, the same pattern of enhancing the CR `status` and annotating the CSV can be used.
+
+``` yaml
+apiVersion:kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+ ...
+status:
+  listeners:
+  - type: plain
+    addresses:
+    - host: myhost1.example.com
+      port: 9092
+    - host: myhost2.example.com
+      port: 9092
+  - type: tls
+    addresses:
+    - host: myhost3.example.com
+      port: 9093
+  bootstrap:
+  - name: plain
+    value: myhost1.example.com:9092,myhost2.example.com:9092
+  - name: tls
+    value: myhost3.example.com
+  caCertificateSecret: my-cluster-cluster-ca-cert
+---
+apiVersion: operators.coreos/com:v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: strimzi-cluster-operator.v0.16.2
+  namespace: placeholder
+  annotations:
+    containerImage: docker.io/strimzi/operator:0.16.2
+spec:
+  customresourcedefinitions:
+    owned:
+    - description: Represents a Kafka cluster
+      displayName: Kafka
+      kind: Kafka
+      name: kafkas.kafka.strimzi.io
+      version: v1beta1
+      statusDescriptors:
+      - description: The secret containing the CA certificate
+        displayName: Secret containing the CA certificate
+        path: status.caCertificateSecret
+        x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:Secret
+        - binding:env:object:secret:ca.p12
```

Comment:
I think this should reference the crt file? The p12 files is really for Java apps only. the CRT file would sound more like Kubernetes style.

Reply:
Yeah it makes sense if we think that on the other side, the application could be written in any other language but not in Java, good point.

Reply:
So refer to all parts of the secret then? That's easy to do and give the most general solution. I'll put that in a revised version and see what you guys think.